screens/Menu/Settings.js

```diff
@@ -55,14 +135,54 @@ function Settings(props) {
                         <Text style={styles.container2SubText}>Current Status is {switchLabel2}</Text>
                         <View style={styles.toggle}>
                             <Switch
-                                value={switchVal2}
-                                onValueChange={(val) => setSwitchVal2(val)}>
+                                value={switchVal2 == "true" ? true : false}
+                                onValueChange={(val) => {
+                                    AsyncStorage.setItem("switchVal", JSON.stringify(val));
+                                    setSwitchVal2(val.toString());
```

Comment:
There is a subtle case for which the user tries to connect, but when doing it, 1. He aborts the log in or 2. It fails ... The toggle stays true but it should go back to false.  
**Possible solution:** I saw that at line 68, you have the case for which the access token is unsuccessful. However, you never using `cancelled` variable. I suggest calling `setSwitchVal2("false")` instead

Reply:
C'est toi le bug hunter hahahah i got u.

Reply:
Fixeddd